lib/ansible/module_utils/docker/swarm.py

```diff
@@ -241,3 +244,11 @@ def get_all_nodes_list(self, output='short'):
 
     def get_node_name_by_id(self, nodeid):
         return self.get_node_inspect(nodeid)['Description']['Hostname']
+
+    def get_unlock_key(self):
+        if self.docker_py_version < LooseVersion('2.7.0'):
+            return None
+        try:
+            return super(AnsibleDockerSwarmClient, self).get_unlock_key()
+        except APIError:
+            return None
```

Comment:
Suppressing errors is usually not a good idea, especially not in library-level code. If this is really necessary, it should be done where this code is called.

Reply:
I agree. Had some though about that it would return 404 on unlocked swarms but it does return an empty string along with a HTTP 200 instead. In `docker_swarm` I think we should catch `APIError` to not fail the entire task just because the key retrieval failed.